src/test/java/plugins/AntPluginTest.java

```diff
@@ -72,6 +99,67 @@ public void locallyInstalledAnt() {
         job.startBuild().shouldSucceed().shouldContainsConsoleOutput(Pattern.quote(expectedVersion));
     }
 
+    @Test @Native("ant")
+    @WithPlugins({"workflow-job", "workflow-cps", "workflow-basic-steps", "workflow-durable-task-step"})
+    public void testAntWrapper() {
+        String antHome = setUpAntInstallation();
+
+        String expectedVersion;
+        if(SystemUtils.IS_OS_WINDOWS) {
+            expectedVersion = jenkins.runScript(String.format("'%s\\bin\\ant -version'.execute().text", antHome));
+        } else {
+            expectedVersion = jenkins.runScript(String.format("'%s/bin/ant -version'.execute().text", antHome));
+        }
+
+        WorkflowJob workflowJob = jenkins.jobs.create(WorkflowJob.class);
+        workflowJob.script.set(SCRIPT_PIPELINE_ANT);
+
+        workflowJob.save();
+
+        workflowJob.startBuild().shouldSucceed();
+
+        String console = workflowJob.getLastBuild().getConsole();
+        assertThat(console, containsString("withAnt"));
+        assertThat(console, containsString("ant -version"));
+        assertThat(console, containsString(expectedVersion));
+    }
+
+    @Test @Native("ant")
+    public void testAdvancedConfiguration() {
+        setUpAnt();
+
+        antBuildStepAdvancedConfiguration(step, BUILD_FILE, PROPERTIES, OPTS);
+
+        job.save();
+
+        job.startBuild().shouldSucceed();
+
+        String console = job.getLastBuild().getConsole();
+        assertThat(console, containsString(System.getProperty("java.version")));
+        assertThat(console, containsString("-D" + OK_PROP1));
+        assertThat(console, containsString("-D" + OK_PROP2));
+        assertThat(console, containsString("-D" + NOK_PROP1));
+        assertThat(console, containsString("-D" + NOK_PROP2));
+        assertThat(console, containsString("[echoproperties] " + OK_PROP1));
+        assertThat(console, containsString("[echoproperties] " + OK_PROP2));
+        assertThat(console, not(Matchers.containsRegexp("[echoproperties] " + NOK_PROP1, Pattern.MULTILINE)));
+        assertThat(console, not(Matchers.containsRegexp("[echoproperties] " + NOK_PROP2, Pattern.MULTILINE)));
+    }
+
+    @Test @Native("ant")
```

Comment:
From [looking at the code](https://github.com/jenkinsci/ant-plugin/blob/master/src/main/java/hudson/tasks/Ant.java#L505), there seems to be a download installer option for Ant. Perhaps the ant setup could be rewritten to use that instead of requiring it on the agent natively?

Reply:
that would change the intent of the test.

You can for always use a docker container with `ant` and `ssh` for the agent. and use a specific node (that will start the container)

e.g. something like https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SshAgentContainer/Dockerfile with the addition of `ant`

https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/main/java/org/jenkinsci/test/acceptance/docker/fixtures/SshAgentContainer.java

Reply:
`SSHAgentContainer` already has `ant` installed 🥳 

https://github.com/jenkinsci/acceptance-test-harness/blob/608f30703d2b8377e251fcb7e2a7637d3c128a2d/src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SshAgentContainer/Dockerfile

https://raw.githubusercontent.com/jenkinsci/docker-fixtures/master/src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/JavaContainer/Dockerfile